### PR TITLE
[FIX] product: empty company record given to currency conversion

### PR DESCRIPTION
**Pull request by FrancescoBallerini**
- Description of the issue/feature this PR addresses:

Calling [price_compute()](https://github.com/odoo/odoo/blob/135c7e1884b3587fcd6e4034a8d2f42d07702106/addons/product/models/product.py#L641) on product.product cause AssertationError **IF you also provide currency parameter.** 

This happens because when currency_id conversion is performed, the product.company_id is False in variants, so maybe it should be investigated why is it false, but this is a very quick fix withoud any drawback.

- Steps to reproduce:

This is very easy to reproduce, but you need a simple snippet because there is no call to price_compute() in product.product while using currency conversion in original source code. Easiest way to reproduce this is to make a button that calls method on a product.product record, you will get AssertationError:

    def btn_test_price_compute(self):

        product = self.env['product.product'].browse(10)  # fetch any product from variant model
        currency = product.currency_id or self.env.company.currency_id  # fetch currency

        price = product.price_compute(
            "list_price",
            uom=product.uom_id,
            currency=currency,
            company=product.company_id
        )[product.id]

        # try to compute price: works fine in product template and also
        # in product product as long as you don't use 'company' argument.
        # As soon as you provide currency a conversion will be performed,
        # but it will fail because there is no product.company_id:
        #
        # >>AssertionError: convert amount from unknown company

- Desired behavior after PR is merged:

We can call price_compute() on product variants using currency conversion parameter without issues.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by Feyensv — COMMENTED** [review]
Indeed, this is wrong and should be fixed :+1:
I've made a little suggestion to improve your solution and harmonize the two `price_compute` methods, please update your commit accordingly and I'll merge the PR.
Thanks

File: addons/product/models/product.py [hunk 1]
```diff
@@ -670,8 +670,9 @@ def price_compute(self, price_type, uom=False, currency=False, company=None):
             # Convert from current user company currency to asked one
             # This is right cause a field cannot be in more than one currency
             if currency:
+                company = product.company_id or self.env.company
```
[the remark this answers is not in the document]
Comment (FrancescoBallerini) [reply addons/product/models/product.py]: Hi @Feyensv , thanks for suggestions! 
I have adapted my commits and CLA file as you asked.

**Comment by FrancescoBallerini** [comment]
> Took the liberty to push on your branch since you had some diff in `product.py` in your CLA commit (which had to be in the other commit) and to improve the commit message as well (follow our [guidelines](https://www.odoo.com/documentation/16.0/contributing/development/git_guidelines.html) and explain the problem with more details).
> 
> @robodoo r+ rebase-ff

I am new to git, thanks for taking care at my place, I appreciate a lot. 
Also I had doubt about being too descriptive, so thanks for clarifying as well. 
I will definitely give a better look into the guidelines for next time 👍

**Comment by Feyensv** [comment]
It's okay, we all went through this learning step.
Thanks for the contribution, have a nice day :).

**Pull request closed without merging**